### PR TITLE
Enable jsdoc eslint rules

### PR DESCRIPTION
**Pull request by jasonnutter**
Adds rules to enforce jsdoc comments. To start, will only enable errors for function parameters.

Note, the rule doesnt always work well with destructor params, so in spots I refactor that to destructure in the function itself.

### CODE REVIEW

File: change/@azure-msal-browser-5961c6d7-83da-4f88-bb97-9ef4f24c9336.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (jasonnutter) [inline change/@azure-msal-browser-5961c6d7-83da-4f88-bb97-9ef4f24c9336.json:2]:
```suggestion
  "type": "none",
```

File: change/@azure-msal-common-62a06824-ad3f-4e6b-b3bd-6e8d99c11389.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (jasonnutter) [inline change/@azure-msal-common-62a06824-ad3f-4e6b-b3bd-6e8d99c11389.json:2]:
```suggestion
  "type": "none",
```

File: package.json [hunk 3]
```diff
@@ -39,7 +40,7 @@
     "test:coverage": "npm run clean:coverage && lerna run test:coverage",
     "test:report": "nyc report",
     "test:e2e": "lerna run --concurrency 1 test:e2e",
-    "lint": "run-s lint:*",
+    "lint": "run-s \"lint:* -- --fix --quiet\"",
```
Comment (jasonnutter) [inline package.json:43]:
```suggestion
    "lint": "run-s lint:*",
```

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 4]
```diff
@@ -502,11 +508,9 @@ export abstract class ClientApplication {
      * MSAL return's a cached token when available
      * Or it send's a request to the STS to obtain a new token using a refresh token.
      *
-     * @param {@link (SilentRequest:type)}
      *
-     * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
+     * @param request
```
Comment (sameerag) [inline lib/msal-browser/src/app/ClientApplication.ts:512]: Why are we losing comments here?

Comment (jasonnutter) [reply lib/msal-browser/src/app/ClientApplication.ts:512]: This comment is no longer true in MSAL Browser.

File: .eslintrc.json [hunk 5]
```diff
@@ -11,15 +11,17 @@
...
     ],
     "extends": [
         "plugin:@typescript-eslint/eslint-recommended",
         "plugin:@typescript-eslint/recommended",
         "plugin:react/recommended",
         "plugin:react-hooks/recommended",
         "plugin:security/recommended",
-        "plugin:import/typescript"
+        "plugin:import/typescript",
+        "plugin:jsdoc/recommended"
```
Comment (jasonnutter) [inline .eslintrc.json:24]:
```suggestion
        "plugin:import/typescript"
```

File: change/@azure-msal-angular-9ee89c06-6de4-4c78-af77-36f614c1b8af.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Enable jsdoc param-name eslint rules",
```
Comment (jasonnutter) [inline change/@azure-msal-angular-9ee89c06-6de4-4c78-af77-36f614c1b8af.json:3]:
```suggestion
  "comment": "Enable jsdoc param-name eslint rules (#2791)",
```

File: change/@azure-msal-browser-5961c6d7-83da-4f88-bb97-9ef4f24c9336.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Enable jsdoc param-name eslint rules",
```
Comment (jasonnutter) [inline change/@azure-msal-browser-5961c6d7-83da-4f88-bb97-9ef4f24c9336.json:3]:
```suggestion
  "comment": "Enable jsdoc param-name eslint rules (#2791)",
```

File: change/@azure-msal-common-62a06824-ad3f-4e6b-b3bd-6e8d99c11389.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Enable jsdoc param-name eslint rules",
```
Comment (jasonnutter) [inline change/@azure-msal-common-62a06824-ad3f-4e6b-b3bd-6e8d99c11389.json:3]:
```suggestion
  "comment": "Enable jsdoc param-name eslint rules (#2791)",
```

File: change/@azure-msal-node-dbffaae5-69be-4331-be1e-fb90571ce7d5.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Enable jsdoc param-name eslint rules",
```
Comment (jasonnutter) [inline change/@azure-msal-node-dbffaae5-69be-4331-be1e-fb90571ce7d5.json:3]:
```suggestion
  "comment": "Enable jsdoc param-name eslint rules (#2791)",
```

File: change/@azure-msal-react-ea219ab9-4047-4729-a551-18b70c65ad7e.json [hunk 10]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Enable jsdoc param-name eslint rules",
```
Comment (jasonnutter) [inline change/@azure-msal-react-ea219ab9-4047-4729-a551-18b70c65ad7e.json:3]:
```suggestion
  "comment": "Enable jsdoc param-name eslint rules (#2791)",
```

File: change/msal-7f7de757-8f00-4084-b64a-fa17c5694cd9.json [hunk 11]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Enable jsdoc param-name eslint rules",
```
Comment (jasonnutter) [inline change/msal-7f7de757-8f00-4084-b64a-fa17c5694cd9.json:3]:
```suggestion
  "comment": "Enable jsdoc param-name eslint rules (#2791)",
```

**Review by jo-arroyo — APPROVED** [review]
Looks good!

File: lib/msal-react/src/components/UnauthenticatedTemplate.tsx [hunk 12]
```diff
@@ -13,7 +13,12 @@ export type UnauthenticatedTemplateProps = PropsWithChildren<AccountIdentifiers>
...
  * Renders child components if user is unauthenticated
- * @param props 
+ *
+ * @param props
+ * @param props.username
+ * @param props.homeAccountId
+ * @param props.localAccountId
+ * @param props.children
  */
 export function UnauthenticatedTemplate({ username, homeAccountId, localAccountId, children }: UnauthenticatedTemplateProps): React.ReactElement|null {
```
Comment (tnorling) [inline lib/msal-react/src/components/UnauthenticatedTemplate.tsx:23]: Do these need to be destructured down below?

File: lib/msal-react/src/components/AuthenticatedTemplate.tsx [hunk 13]
```diff
@@ -13,7 +13,12 @@ export type AuthenticatedTemplateProps = PropsWithChildren<AccountIdentifiers>;
...
  * Renders child components if user is authenticated
- * @param props 
+ *
+ * @param props
+ * @param props.username
+ * @param props.homeAccountId
+ * @param props.localAccountId
+ * @param props.children
  */
 export function AuthenticatedTemplate({ username, homeAccountId, localAccountId, children }: AuthenticatedTemplateProps): React.ReactElement|null {
```
Comment (tnorling) [inline lib/msal-react/src/components/AuthenticatedTemplate.tsx:23]: Do these need to be destructured down below?

**Review by pkanher617 — APPROVED** [review]
Would like to resolve the comments from @tnorling but otherwise looks good.

**Pull request closed without merging**